pydantic_ai_slim/pydantic_ai/_tool_manager.py

```diff
@@ -121,7 +134,15 @@ async def _call_tool(self, call: ToolCallPart, allow_partial: bool, wrap_validat
             else:
                 args_dict = validator.validate_python(call.args or {}, allow_partial=pyd_allow_partial)
 
-            return await self.toolset.call_tool(name, args_dict, ctx, tool)
+            if usage_limits is not None:
+                usage_limits.check_before_tool_call(self.ctx.usage)
+
+            result = await self.toolset.call_tool(name, args_dict, ctx, tool)
+
+            if usage_limits is not None:
+                self.ctx.usage.tool_calls += 1
```

Comment:
I'm realizing now that the `usage_limits` check here doesn't make total sense, because if this is a non-output tool but usage limits are not configured, we'd still want the `usage.tool_calls` to have the correct value in case the user wants to check it themselves...

What do you think of putting this incrementation after the call to `self._call_tool_traced` in `handle_call`?